suite-native/passphrase/src/useFeatureRequestsPassphrase.tsx

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import TrezorConnect from '@trezor/connect';
+import {
+    PassphraseStackParamList,
+    PassphraseStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import {
+    deviceActions,
+    selectDevice,
+    selectDeviceButtonRequestsCodes,
+} from '@suite-common/wallet-core';
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    PassphraseStackParamList,
+    PassphraseStackRoutes,
+    RootStackParamList
+>;
+
+// When device is in view only mode and is reconnected, passphrase is not requested after reconnecting
+// but when the app interacts with the device (e.g. requesting address / validating transaction, etc).
+// To make the passphrase working on this feature, just use this hook in the component that will interact with the device
+// and add the features button request code to the useEffect that redirects back to the flow
+// after successfull passphrase submission.
+export const useFeatureRequestsPassphrase = () => {
+    const dispatch = useDispatch();
+
+    const navigation = useNavigation<NavigationProp>();
+
+    const device = useSelector(selectDevice);
+    const buttonRequestCodes = useSelector(selectDeviceButtonRequestsCodes);
+
+    useEffect(() => {
+        if (
+            buttonRequestCodes.includes('ButtonRequest_Address') &&
+            buttonRequestCodes.includes('ButtonRequest_Other') &&
+            navigation.canGoBack()
+        ) {
+            navigation.goBack();
```

Comment:
I don't think this is a good place to navigate back. Closing passphrase screen can be hopefully handled inside passhprase screen? This seems to be fragile, not sure on what screen it happens and what is back. It might navigate back to PIN matrix screen.

Reply:
TT, passphrase on device, I got stucked with this screen after passphrase confirmation on the device and device also ask me to verify address on this screen
<img width="350" alt="image" src="https://github.com/trezor/trezor-suite/assets/3729633/5f16a97f-e812-4daf-90a8-d7767ce83720">

Reply:
demonstration of goBack on PIN screen

https://github.com/trezor/trezor-suite/assets/3729633/e8e93f82-50cc-4dfa-8098-96122418d4f6